project/src/main/puzzle/milestone-hud.gd

```diff
@@ -45,20 +45,23 @@ func _process(_delta: float) -> void:
 
 ## Updates the milestone progress bar's value and boundaries.
 func update_milebar_values() -> void:
-	_progress_bar.min_value = MilestoneManager.prev_milestone().value
+	_progress_bar.min_value = MilestoneManager.prev_milestone().adjusted_value()
 	var next_milestone := MilestoneManager.next_milestone()
-	if next_milestone.value == _progress_bar.min_value:
+	var next_milestone_value := next_milestone.adjusted_value()
```

Comment:
Don't store this.

Reply:
next_milestone.adjusted_value() is now referenced in conditionals and not stored.